### PR TITLE
FOF virtual file fix

### PR DESCRIPTION
**Pull request by robjmcgibbon**
Changes
- For DMO runs we want to use the FOF IDs from the snapshots, but I was updating the virtual source to always point to the membership file.
- When we create virtual files I am adding a field called `HaloCatalogueIndex`, which is just a link to `GroupNr_bound`.
- Make some minor corrections to the docs

### CODE REVIEW

File: documentation/SOAP.tex [hunk 1]
```diff
@@ -223,7 +223,7 @@ \section{Spherical overdensity calculations}
 \begin{enumerate}
     \item Sort particles according to radius and construct the cumulative mass profile.
     \item Discard any particles at zero radius, since we cannot compute a density for those. The mass of these 
-    particles is used as an $r=0$ offset for the cumulative mass profile. Since the centre of potential is the 
+    particles is used as an $r=0$ offset for the cumulative mass profile. Since the halos centre is the
```
Comment (VictorForouhar) [inline documentation/SOAP.tex:226]: typo: halo's centre OR centre of the halo.

File: documentation/footnote_AngMom.tex [hunk 2]
```diff
@@ -1,5 +1,5 @@
 \paragraph{$^{$FOOTNOTE_NUMBER$}$The angular momentum} of gas, dark matter and stars is computed relative to 
-the centre of potential (cop) and the centre of mass velocity of that particular component, and not to to the 
+the halo centre (cop) and the centre of mass velocity of that particular component, and not to to the
```
Comment (VictorForouhar) [inline documentation/footnote_AngMom.tex:2]: Repeated to towards the end of the line: "... and not to to the "

**Comment by robjmcgibbon** [comment]
Script for generating a virtual file
```snapshot="/cosma8/data/dp004/flamingo/Runs/L1000N1800/HYDRO_FIDUCIAL/snapshots/flamingo_{snap_nr:04}/flamingo_{snap_nr:04}.hdf5"
membership="/cosma8/data/dp004/flamingo/Runs/L1000N1800/HYDRO_FIDUCIAL/SOAP-HBT/membership_{snap_nr:04}/membership_{snap_nr:04}.{file_nr}.hdf5"
output_file="/snap8/scratch/dp004/dc-mcgi1/virtual/L1000N1800/HYDRO_FIDUCIAL/flamingo_{snap_nr:04}.hdf5"
snapnum=77
mkdir -p $(dirname $output_file)
python make_virtual_snapshot.py $snapshot $membership $output_file $snapnum

**Comment by VictorForouhar** [comment]
Ran the script successfully. All looks good.

**Pull request merged**